tic-tac-toe.js

```diff
@@ -1,7 +1,157 @@
+TicTacToe.prototype.begin = function(){
+  this.updateBoard([[0,0,0], [0,0,0],[0,0,0]]);
+  this.updatePlayer("trooper");
+  this.updateWinner(false);
+  $(".gameover").addClass("hide");
+  $(".gameover").children("p").addClass("hide");
+  $("td").removeClass("vader trooper");
+};
+
+
+//logic for a single turn
+TicTacToe.prototype.takeTurn = function(clickedSquare){
+  var coords = this.getCoords(clickedSquare);
+  var nextBoard = this.markArray(coords);
+  this.updateBoard(nextBoard);
+  this.markSquare(clickedSquare);
+  this.checkGame(); //checks to see if game is won
+  if (this.currentWinner !== false){
+    this.endGame(this.currentWinner);
+  } else {
+    this.changePlayer();
+  }
+};
+
+
+//defines who the current player is
+TicTacToe.prototype.updatePlayer = function(nextPlayer){
+  this.currentPlayer = nextPlayer;
+};
+
+//defines who the current winner is
+TicTacToe.prototype.updateWinner = function(nextWinner){
+  this.currentWinner = nextWinner;
 }
 
-TicTacToe.prototype = {
-  
+//defines what the current board is
+TicTacToe.prototype.updateBoard = function(nextBoard) {
+  this.currentBoard = nextBoard;
 }
+
+//gets the coordinates from which square was clicked
+TicTacToe.prototype.getCoords = function(square) {
+  var rowInd = (square.parentNode).rowIndex;
+  var colInd = square.cellIndex;
+  var coordinates = {row: rowInd, col: colInd};
+  return coordinates
+};
+
+//checks to see if game is finished
+TicTacToe.prototype.checkGame = function () {
+  var board = this.currentBoard;
+  var win = false;
+
+  //identify combos that need to be checked
+  //situation is simply enough that we don't really need to iterate
+  var row1 = board[0];
+  var row2 = board[1];
+  var row3 = board[2];
+  var col1 = [row1[0], row2[0], row3[0]];
+  var col2 = [row1[1], row2[1], row3[1]];
+  var col3 = [row1[2], row2[2], row3[2]];
+  var diag1 = [row1[0], row2[1], row3[2]];
+  var diag2 = [row1[2], row2[1], row3[0]];
+
+  //initialize an array that will store combos to be checked
+  var combos = [row1, row2, row3, col1, col2, col3, diag1, diag2]
+
+  //get rid of any combo that has a zero in it
+  var nonzeroCombos = new Array;
+  combos.forEach(function(combo) {
+    if (combo[0] != 0 && combo[1] != 0 && combo[2] != 0) {
+      nonzeroCombos.push(combo);
+    }
+    return nonzeroCombos;
+  });
+
+  //check to see if any combos contain all the same value
+  nonzeroCombos.forEach(function(combo) {
+      if (combo[0] === combo[1] && combo[1] === combo[2]) {
+        win = true;
+      }
+    return win;
+   });
+
+ if (win == true) {
+   this.updateWinner(this.currentPlayer);
+ }
+
+ //if there are no combos with zero, and no winner, the board must be full
+ if (nonzeroCombos.length == 8 && this.currentWinner == false) {
+  this.updateWinner("tie");
+ }
+};
+
+//changes display when the game is over
+//new game isn't started until the next click on a box
+TicTacToe.prototype.endGame = function (winner) {
```

Comment:
`winner` is passed in as an argument but is never used. Did you intent to use it?